File: Library.ViewModels/BookViewModels/CreateBookViewModel.cs
using Library.ViewModels.AuthorViewModels;
using Library.ViewModels.PublicationHouseViewModels;
using System.Collections.Generic;

namespace Library.ViewModels.BookViewModels
{
    public class CreateBookViewModel
    {
		public long Id { get; set; }
		public string Name { get; set; }
		public int YearOfPublication { get; set; }
		public long? AuthorId { get; set; }
		public GetAuthorViewModel Author { get; set; }
		public List<GetPublicationHouseViewModel> PublicationHouses { get; set; }
	}
}

File: Library.WebUI/src/app/models/book/CreateBookViewModel.ts
import { GetAuthorViewModel } from "../author/GetAuthorViewModel";
import { GetPublicationHouseViewModel } from "../publicationHouse/GetPublicationHouseViewModel";

export class CreateBookViewModel {
    constructor(
        public id?: number,
        public name?: string,
        public yearOfPublication?: number,

        public author?: GetAuthorViewModel,
        public publicationHouses?: GetPublicationHouseViewModel[]) { }
}
